### PR TITLE
Initial cleanup of the unsupported platforms and installers 

### PR DESCRIPTION
**Pull request by ibuziuk**
### What does this PR do?
-  Removing unsupported installers (helm & minishift addon)
-  Removing unsupported platforms: k8s, docker-desktop, microk8s, minikube, minishift 
-  openshift & crc are the only supported platforms & operator is the only supported installer type

### What issues does this PR fix or reference?
https://issues.jboss.org/browse/CRW-463

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -25,8 +25,6 @@
     "cli-ux": "^5.3.3",
     "command-exists": "^1.2.8",
     "debug": "^4.1.1",
-    "eclipse-che": "git://github.com/eclipse/che#7.3.1",
-    "eclipse-che-minishift": "git://github.com/minishift/minishift#master",
     "eclipse-che-operator": "git://github.com/redhat-developer/codeready-workspaces-operator#master",
```
Comment (benoitf) [inline package.json]: we should rename eclipse-che-operator to codeready-workspaces-operator

File: package.json [hunk 2]
```diff
@@ -120,12 +118,10 @@
...
   "repository": "che-incubator/chectl",
   "scripts": {
-    "postinstall": "npm run -s postinstall-repositories && npm run -s postinstall-helm && npm run -s postinstall-operator && npm run -s postinstall-minishift-addon && npm run -s postinstall-cleanup",
-    "postinstall-helm": "rimraf templates/kubernetes && cpx 'node_modules/eclipse-che/deploy/kubernetes/**' 'templates/kubernetes'",
-    "postinstall-minishift-addon": "rimraf templates/minishift-addon && cpx 'node_modules/eclipse-che-minishift/addons/che/**' 'templates/minishift-addon/che'",
+    "postinstall": "npm run -s postinstall-repositories && npm run -s postinstall-operator && npm run -s postinstall-cleanup",
     "postinstall-operator": "rimraf templates/che-operator && cpx 'node_modules/eclipse-che-operator/deploy/**' 'templates/che-operator'",
-    "postinstall-repositories": "yarn upgrade eclipse-che eclipse-che-operator eclipse-che-minishift",
-    "postinstall-cleanup": "rimraf node_modules/eclipse-che && rimraf node_modules/eclipse-che-operator && rimraf node_modules/eclipse-che-minishift",
+    "postinstall-repositories": "yarn upgrade eclipse-che-operator",
```
Comment (benoitf) [inline package.json]: as said before I think the operator should be renamed codeready-workspaces-operator for better readability

File: src/commands/server/start.ts [hunk 3]
```diff
@@ -227,7 +197,7 @@ export default class Start extends Command {
         || (ctx.isKeycloakDeployed && !ctx.isKeycloakReady)
         || (ctx.isPluginRegistryDeployed && !ctx.isPluginRegistryReady)
         || (ctx.isDevfileRegistryDeployed && !ctx.isDevfileRegistryReady)) {
-        if (flags.platform || flags.installer) {
+        if (flags.platform) {
           this.warn('Deployed Che is found and the specified installation parameters will be ignored')
```
Comment (benoitf) [inline src/commands/server/start.ts]: for later and not related to this PR, but we should probably introduce in upstream chectl APP_NAME or PRODUCT_NAME else for example we will see 'Che' name appearing

File: src/tasks/platforms/openshift.ts [hunk 4]
```diff
@@ -13,7 +13,32 @@ import * as commandExists from 'command-exists'
...
+      task: async (ctx: any, task: any) => {
+        try {
+          await kube.checkKubeApi()
+          ctx.isOpenShift = await kube.isOpenShift()
+          task.title = await `${task.title}...OK`
+          if (ctx.isOpenShift) {
+            task.title = await `${task.title} (it's OpenShift)`
+          }
+        } catch (error) {
+          command.error(`Failed to connect to Kubernetes API. ${error.message}`)
```
Comment (benoitf) [inline src/tasks/platforms/openshift.ts]: Kubernetes --> OpenShift

File: package.json (hunk 1 above)
Comment (ibuziuk) [reply package.json]: yeah,  will do it as part of the renaming subtask of `cheсtl` to `crwctl`

File: package.json (hunk 2 above)
Comment (ibuziuk) [reply package.json]: ack

File: src/commands/server/start.ts (hunk 3 above)
Comment (ibuziuk) [reply src/commands/server/start.ts]: yeah, It is an important discussion for future

File: src/tasks/platforms/openshift.ts (hunk 4 above)
Comment (ibuziuk) [reply src/tasks/platforms/openshift.ts]: will do as part of renaming sub-task

**Pull request merged**